external-applications/quarkus-workshop-super-heroes/src/test/java/io/quarkus/ts/openshift/heroes/workshop/HeroesOpenShiftIT.java

```diff
@@ -0,0 +1,193 @@
+        Hero hero = new Hero();
+        hero.name = DEFAULT_NAME;
+        hero.otherName = DEFAULT_OTHER_NAME;
+        hero.level = DEFAULT_LEVEL;
+        hero.picture = DEFAULT_PICTURE;
+        hero.powers = DEFAULT_POWERS;
+
+        String location = given()
+            .body(hero)
+            .header(CONTENT_TYPE, APPLICATION_JSON)
+            .header(ACCEPT, APPLICATION_JSON)
+            .when()
+            .post(url + "/api/heroes")
+            .then()
+            .statusCode(CREATED.getStatusCode())
+            .extract().header("Location");
+        assertTrue(location.contains("/api/heroes"));
+
+
+        String[] segments = location.split("/");
+        heroId = segments[segments.length - 1];
+        assertNotNull(heroId);
+
+        given()
+            .pathParam("id", heroId)
+            .when().get("/api/heroes/{id}")
+            .then()
+            .statusCode(OK.getStatusCode())
+            .header(CONTENT_TYPE, APPLICATION_JSON)
+            .body("name", is(DEFAULT_NAME))
+            .body("otherName", is(DEFAULT_OTHER_NAME))
+            .body("level", is(DEFAULT_LEVEL*3))
+            .body("picture", is(DEFAULT_PICTURE))
+            .body("powers", is(DEFAULT_POWERS));
+    }
+
+    @Test
+    @Order(2)
+    public void testUpdateHero() {
+        Hero hero = new Hero();
+        hero.id = new Long(heroId);
+        hero.name = UPDATED_NAME;
+        hero.otherName = UPDATED_OTHER_NAME;
+        hero.level = UPDATED_LEVEL;
+        hero.picture = UPDATED_PICTURE;
+        hero.powers = UPDATED_POWERS;
+
+        given()
+            .body(hero)
+            .header(CONTENT_TYPE, APPLICATION_JSON)
+            .header(ACCEPT, APPLICATION_JSON)
+            .when()
+            .put("/api/heroes")
+            .then()
+            .statusCode(OK.getStatusCode())
+            .header(CONTENT_TYPE, APPLICATION_JSON)
+            .body("name", is(UPDATED_NAME))
+            .body("otherName", is(UPDATED_OTHER_NAME))
+            .body("level", is(UPDATED_LEVEL))
+            .body("picture", is(UPDATED_PICTURE))
+            .body("powers", is(UPDATED_POWERS));
+    }
+
+    @Test
+    @Order(3)
+    public void testDeleteHero() {
+        given()
+            .pathParam("id", heroId)
+            .when().delete("/api/heroes/{id}")
+            .then()
+            .statusCode(NO_CONTENT.getStatusCode());
+    }
+
+    @Test
+    @Order(4)
+    public void testCalledOperationMetrics() {
+        given()
+            .header(ACCEPT, APPLICATION_JSON)
+            .when().get(url + "/metrics/application")
+            .then()
+            .statusCode(OK.getStatusCode())
+            .body("'io.quarkus.workshop.superheroes.hero.HeroResource.countCreateHero'", is(1))
+            .body("'io.quarkus.workshop.superheroes.hero.HeroResource.countUpdateHero'", is(1))
+            .body("'io.quarkus.workshop.superheroes.hero.HeroResource.countDeleteHero'", is(1));
+    }
+
+    class Hero {
```

Comment:
This should be a `static class`, it doens't need to hold a reference to the test class.